pkg/utils/addons/addons.go

```diff
@@ -121,26 +142,65 @@ func CreatePipelines(template mf.Manifest, client client.Client) (mf.Manifest, e
 			pipelineParams = append(pipelineParams.([]interface{}), map[string]interface{}{"name": "MINOR_VERSION", "type": "string"})
 		}
 		if err := unstructured.SetNestedField(newTempRes.Object, pipelineParams, "spec", "params"); err != nil {
-			return mf.Manifest{}, err
+			return nil, err
 		}
 
 		if err := unstructured.SetNestedField(taskBuild, taskParams, "params"); err != nil {
-			return mf.Manifest{}, nil
+			return nil, nil
 		}
 
 		//adding the deploy task
 		for _, tg := range taskGenerators {
 			p, err := tg.generate(newTempRes)
 			if err != nil {
-				return mf.Manifest{}, err
+				return nil, err
 			}
 			pipelines = append(pipelines, p)
 		}
+	}
+	return pipelines, nil
+}
+
+func CreatePipelines(templatePath string, client client.Client) (mf.Manifest, error) {
+	var pipelines []unstructured.Unstructured
+
+	workspacedTemplate, err := mf.NewManifest(path.Join(templatePath, "pipeline_using_workspace.yaml"))
+	if err != nil {
+		return mf.Manifest{}, err
+	}
 
+	workspacedTaskGenerators := []taskGenerator{
+		&pipeline{environment: "openshift", nameSuffix: "", generateDeployTask: openshiftDeployTask},
+		&pipeline{environment: "kubernetes", nameSuffix: "-deployment", generateDeployTask: kubernetesDeployTask},
+		&pipeline{environment: "knative", nameSuffix: "-knative", generateDeployTask: knativeDeployTask},
 	}
-	manifests, err := mf.ManifestFrom(mf.Slice(pipelines), mf.UseClient(mfc.NewClient(client)))
+
+	wps, err := generateBasePipeline(workspacedTemplate, workspacedTaskGenerators, !resourced)
+	if err != nil {
+		return mf.Manifest{}, err
+	}
+	pipelines = append(pipelines, wps...)
+
+	resourcedTemplate, err := mf.NewManifest(path.Join(templatePath, "pipeline_using_resource.yaml"))
+	if err != nil {
+		return mf.Manifest{}, err
+	}
+
+	resourcedTaskGenerators := []taskGenerator{
+		&pipeline{environment: "openshift", nameSuffix: "", resourcedSuffix: "-pr", generateDeployTask: openshiftDeployTask},
+		&pipeline{environment: "kubernetes", nameSuffix: "-deployment", resourcedSuffix: "-pr", generateDeployTask: kubernetesDeployTask},
+		&pipeline{environment: "knative", nameSuffix: "-knative", resourcedSuffix: "-pr", generateDeployTask: knativeResourcedDeployTask},
```

Comment:
why is only knative pipline needs a `knativeResourcedDeployTask` where as the other 2 seems to be reusing the `openshiftDepllyTask` and `kubernetesDeployTask`

Reply:
deploy task for openshift, kubernetes environment does not have any change using resource or using the workspace, but for knative there is.